### PR TITLE
MS BFS python APIs + EgoNet updates

### PR DESCRIPTION
**Pull request by afender**
There are various things in this PR.

Multi-Seed (MS) BFS: 
- API tentative 
- Saving research on memory feasability helper function (not in production)
- Saving research on running the current BFS concurrently with streams and threads for analysis perf comparison (not in production)

EgoNet:
- Multithreading in EgoNet which deserializes execution and comes with mild performance improvements on large sizes
- Some cleanup

### CODE REVIEW

**Review by ChuckHastings — APPROVED** [review]

File: python/cugraph/traversal/ms_bfs.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.
```
Comment (Iroy30) [inline python/cugraph/traversal/ms_bfs.py:1]: Do we want a test for this?

Comment (Iroy30) [reply python/cugraph/traversal/ms_bfs.py:1]: I see that it is not been implemented but exposed, is that right?
In that case, we could either raise error or not expose it to user

Comment (afender) [reply python/cugraph/traversal/ms_bfs.py:1]: Right, it is just the API proposal. I can raise an error.

**Review by Iroy30 — APPROVED** [review]

**Review by BradReesWork — APPROVED** [review]

**Comment by BradReesWork** [comment]
@gpucibot merge

**Pull request merged**